### DISCUSSION TITLE
Split image with text into 2 files, between 2 lines

### DISCUSSION THREAD

**Discussion by mikolajek-fr** [ImageMagick/ImageMagick#2509] [Help]
Hi,

I'm looking for a solution about horizontaly splitting an image into 2 files around 50% each, but :

• Image is a text list,
• The split must be between lines,
• All the text has the same size,
• No blank line between two lines.

Example : 

![Sans titre](https://user-images.githubusercontent.com/24416331/92254557-5d5a5c80-eed1-11ea-9b90-2ef0113f64b0.png)

Thanks for your help !

**Comment by snibgo** [comment]
What version of IM, on what platform? I will assume v7, on Windows BAT. This breaks at the first white space on or after the middle.

```
set INFILE=textLines.png

for /F "usebackq" %%L in (`%IMG7%magick ^
  %INFILE% ^
  -scale "1x%%[fx:h]^!" ^
  -fill Black +opaque White ^
  -colorspace Gray ^
  -crop 1x2@ +repage ^
  ^( -clone 0 ^
     -format "HH1=%%h\n" ^
     +write info: ^
     +delete ^
  ^) ^
  -delete 0 ^
  ^( -clone 0 ^
     -bordercolor black -border 1 ^
     -trim ^
     -format "SECND=%%Y\n" ^
     +write info: ^
     +delete ^
  ^) ^
  NULL:`) do set %%L

set /A CutAt=%HH1%+(%SECND%)-1

echo Cut at y=%CutAt%

%IMG7%magick ^
  %INFILE% ^
  -crop x%CutAt% ^
  out-%%02d.png
```
out-00.png:
![out-00](https://user-images.githubusercontent.com/33812368/92270101-59ccd280-eedd-11ea-9933-480cb0957c5b.png)

out-01.png:
![out-01](https://user-images.githubusercontent.com/33812368/92270122-60f3e080-eedd-11ea-956e-1f3f0d337685.png)

**Reply by mikolajek-fr** [reply]
Is it possible to have a bash version of this script ? Converting batch / bash is a bit difficult.

Thanks :)

**Comment by mikolajek-fr** [comment]
Thanks, that's it : perfectly splitted in the middle !

IM 7 on Linux :) Is it easily adaptable for bash ?

**Comment by snibgo** [comment]
Translation to bash should be very simple.
```
INFILE=textLines.png

declare `magick \
  ${INFILE} \
  -scale "1x%[fx:h]!" \
  -fill Black +opaque White \
  -colorspace Gray \
  -crop 1x2@ +repage \
  \( -clone 0 \
     -format "HH1=%h\n" \
     +write info: \
     +delete \
  \) \
  -delete 0 \
  \( -clone 0 \
     -bordercolor black -border 1 \
     -trim \
     -format "SECND=%Y\n" \
     +write info: \
     +delete \
  \) \
  NULL:`

CutAt=$(echo "${HH1}${SECND}-1" | bc)

echo Cut at y=${CutAt}

magick \
  ${INFILE} \
  -crop x${CutAt} \
  out-%02d.png
```
